### PR TITLE
build: enable useUnknownInCatchVariables for packages

### PR DESCRIPTION
**Pull request by atscott**
This unblocks the internal migration to turn the option on in g3.

### CODE REVIEW

File: packages/tsconfig.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "outDir": "../dist/all/@angular",
     "noImplicitAny": true,
     "noImplicitOverride": true,
-    "useUnknownInCatchVariables": false,
+    "useUnknownInCatchVariables": true,
```
Comment (alfaproject) [inline packages/tsconfig.json:20]: You can just remove this now. It's part of `strict` I believe?

**Comment by atscott** [comment]
[presubmit](https://fusion2.corp.google.com/presubmit/tap/422578754/OCL:422578754:BASE:422578827:1642527876641:804c068b/targets)

**Comment by dylhunn** [comment]
Labeling this as core to clear it from the triage queue, it doesn't really fit in any single component though.

**Review by alxhub — CHANGES_REQUESTED** [review]
We should also update `tsconfig-build.json` files as these explicitly override `useUnknownInCatchVariables` to be `false`.

**Review by alxhub — APPROVED** [review]

**Comment by atscott** [comment]
[presubmit](https://fusion2.corp.google.com/presubmit/tap/421419430/OCL:421419430:BASE:425466167:1643669128016:1aa6c385/targets)

**Review by josephperrott — APPROVED** [review]
LGTM

Approved-for: global-approvers

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit fdfcef5a0a6187eeffa936de86d780d464178521.

**Pull request closed without merging**